app/api/entities/entitySchema.js

```diff
@@ -0,0 +1,304 @@
+          }
+        },
+        {
+          properties: {
+            from: { type: 'number' },
+            to: { type: 'null' }
+          }
+        }
+      ]
+    }
+  },
+  properties: {
+    _id: { type: 'string', minLength: 1 },
+    sharedId: { type: 'string', minLength: 1 },
+    language: { type: 'string', minLength: 1 },
+    mongoLanguage: { type: 'string' },
+    title: { type: 'string', minLength: 1 },
+    template: { type: 'string', minLength: 1 },
+    file: {
+      type: 'object',
+      properties: {
+        originalname: { type: 'string' },
+        filename: { type: 'string' },
+        mimetype: { type: 'string' },
+        size: { type: 'number' },
+        timestamp: { type: 'number' },
+        language: { type: 'string' }
+      }
+    },
+    fullText: {
+      type: 'object',
+      patternProperties: {
+        '^[0-9]+$': { type: 'string' }
+      }
+    },
+    totalPages: { type: 'number' },
+    icon: {
+      type: 'object',
+      properties: {
+        _id: { type: 'string' },
+        label: { type: 'string' },
+        type: { type: 'string' }
+      }
+    },
+    toc: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          label: { type: 'string' },
+          indentation: { type: 'number' },
+          range: {
+            type: 'object',
+            properties: {
+              start: { type: 'number' },
+              end: { type: 'number' }
+            }
+          }
+        }
+      }
+    },
+    attachments: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          originalname: { type: 'string' },
+          filename: { type: 'string' },
+          mimetype: { type: 'string' },
+          timestamp: { type: 'number' },
+          size: { type: 'number' }
+        }
+      }
+    },
+    creationDate: { type: 'number' },
+    processed: { type: 'boolean' },
+    uploaded: { type: 'boolean' },
+    published: { type: 'boolean' },
+    pdfInfo: {
+      type: 'object',
+      patternProperties: {
+        '^[0-9]+$': {
+          type: 'object',
+          properties: {
+            chars: { type: 'number' }
+          }
+        }
+      }
+    },
+    user: { type: 'string' },
+    metadata: {
+      type: 'object',
+      additionalProperties: {
+        oneOf: [
+          { type: 'string' },
+          { type: 'number' },
+          {
+            type: 'array',
+            items: {
+              type: 'string'
+            }
+          },
+          {
+            type: 'array',
+            items: {
+              type: 'number'
+            }
+          },
+          {
+            $ref: '#/definitions/dateRange'
+          },
+          {
+            type: 'array',
+            items: {
+              $ref: '#/definitions/dateRange'
+            }
+          },
+          {
+            type: 'object',
+            required: ['label', 'url'],
+            properties: {
```

Comment:
this could be an extracted definitions/linkFieldSchema

Reply:
A separate const like other suggestions? Or a definition embedded in the main schema?

Reply:
I would prefer to separate out all definitions generally.